<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Evgenii Tretiakov" />

<meta name="date" content="2021-01-28" />

<title>Onecut3 in mouse hypothalamus development</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Zupancic_2022</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="dynamic_oc3-pop.html">Glutamate and GABA dynamic in Onecut3 neurons</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Onecut3 in mouse hypothalamus development</h1>
<h4 class="author">Evgenii Tretiakov</h4>
<h4 class="date">2021-01-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Zupancic_2022/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220105code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220105)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220105code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220105)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomEugOTZupancic2022tree1df6d48261e1a1c70a011db219336c3ce3168c69targetblank1df6d48a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/EugOT/Zupancic_2022/tree/1df6d48261e1a1c70a011db219336c3ce3168c69" target="_blank">1df6d48</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomEugOTZupancic2022tree1df6d48261e1a1c70a011db219336c3ce3168c69targetblank1df6d48a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/EugOT/Zupancic_2022/tree/1df6d48261e1a1c70a011db219336c3ce3168c69" target="_blank">1df6d48</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/oldCCA_nae_srt.rds
    Ignored:    data/rar2020.srt.cont.oc2or3.raw.h5seurat
    Ignored:    data/rar2020.srt.neuro.raw.h5seurat
    Ignored:    output/tables/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/dynamic_oc3-pop.Rmd</code>) and HTML (<code>docs/dynamic_oc3-pop.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/EugOT/Zupancic_2022/blob/1df6d48261e1a1c70a011db219336c3ce3168c69/analysis/dynamic_oc3-pop.Rmd" target="_blank">1df6d48</a>
</td>
<td>
EugOT
</td>
<td>
2022-01-29
</td>
<td>
wflow_publish(“*“)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/EugOT/Zupancic_2022/d106e47185a9b836156cab167df844bb9d1b3ebb/docs/dynamic_oc3-pop.html" target="_blank">d106e47</a>
</td>
<td>
EugOT
</td>
<td>
2022-01-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/EugOT/Zupancic_2022/blob/6a77e95d62e71bb92e766f552b45cccfc254c23e/analysis/dynamic_oc3-pop.Rmd" target="_blank">6a77e95</a>
</td>
<td>
EugOT
</td>
<td>
2022-01-29
</td>
<td>
wflow_publish(“*“)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/EugOT/Zupancic_2022/6a77e95d62e71bb92e766f552b45cccfc254c23e/docs/dynamic_oc3-pop.html" target="_blank">6a77e95</a>
</td>
<td>
EugOT
</td>
<td>
2022-01-29
</td>
<td>
wflow_publish(“*“)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code># Load tidyverse infrastructure packages
library(here)
# here() starts at /home/etretiakov/src/Zupancic_2022
library(tidyverse)
# ── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──
# ✔ ggplot2 3.3.5     ✔ purrr   0.3.4
# ✔ tibble  3.1.6     ✔ dplyr   1.0.7
# ✔ tidyr   1.1.4     ✔ stringr 1.4.0
# ✔ readr   2.1.1     ✔ forcats 0.5.1
# ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
# ✖ dplyr::filter() masks stats::filter()
# ✖ dplyr::lag()    masks stats::lag()
library(magrittr)
# 
# Attaching package: &#39;magrittr&#39;
# The following object is masked from &#39;package:purrr&#39;:
# 
#     set_names
# The following object is masked from &#39;package:tidyr&#39;:
# 
#     extract
library(zeallot)
library(future)
# 
# Attaching package: &#39;future&#39;
# The following objects are masked from &#39;package:zeallot&#39;:
# 
#     %-&gt;%, %&lt;-%

# Load packages for scRNA-seq analysis and visualisation
library(sctransform)
library(Seurat)
# Registered S3 method overwritten by &#39;spatstat.geom&#39;:
#   method     from
#   print.boxx cli
# Attaching SeuratObject
library(SeuratWrappers)
library(SeuratDisk)
# Registered S3 method overwritten by &#39;SeuratDisk&#39;:
#   method            from  
#   as.sparse.H5Group Seurat
library(UpSetR)
library(patchwork)
library(Nebulosa)</code></pre>
<pre class="r"><code>src_dir    &lt;- here(&quot;code&quot;)
data_dir   &lt;- here(&quot;data&quot;)
output_dir &lt;- here(&quot;output&quot;)
plots_dir  &lt;- here(output_dir, &quot;figures&quot;)
tables_dir &lt;- here(output_dir, &quot;tables&quot;)</code></pre>
<pre class="r"><code>source(here(src_dir, &quot;functions.R&quot;))
source(here(src_dir, &quot;genes.R&quot;))</code></pre>
<pre class="r"><code>reseed &lt;- 42
set.seed(seed = reseed)</code></pre>
<pre class="r"><code># available cores
n_cores &lt;- available_cores(prop2use = .5)
# Parameters for parallel execution
plan(&quot;multicore&quot;, workers = n_cores)
options(future.globals.maxSize = Inf,
        future.rng.onMisuse = &quot;ignore&quot;)
plan()
# multicore:
# - args: function (..., envir = parent.frame(), workers = 16)
# - tweaked: TRUE
# - call: plan(&quot;multicore&quot;, workers = n_cores)</code></pre>
<p>Read data</p>
<pre class="r"><code>rar2020_ages_all     &lt;- c(&quot;E15&quot;, &quot;E17&quot;, &quot;P00&quot;, &quot;P02&quot;, &quot;P10&quot;, &quot;P23&quot;)
rar2020_ages_postnat &lt;-                      c(&quot;P02&quot;, &quot;P10&quot;, &quot;P23&quot;)
samples_df &lt;- read_tsv(here(&quot;data/samples.tsv&quot;))
# Rows: 8 Columns: 14
# ── Column specification ────────────────────────────────────────────────────────
# Delimiter: &quot;\t&quot;
# chr (8): sample, age, condition, fullname, name, sex, date, date10x
# dbl (6): ncells, libbatch, seqbatch, perfussed, nt, sn
# 
# ℹ Use `spec()` to retrieve the full column specification for this data.
# ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
colours_wtree &lt;- setNames(read_lines(here(data_dir, &quot;colours_wtree.tsv&quot;)),
                          1:45)

onecut  &lt;- LoadH5Seurat(here(data_dir, &quot;rar2020.srt.cont.oc2or3.raw.h5seurat&quot;))
# Validating h5Seurat file
# Initializing RNA with data
# Adding counts for RNA
# Adding scale.data for RNA
# Adding variable feature information for RNA
# Adding reduction pca
# Adding cell embeddings for pca
# Adding feature loadings for pca
# Adding miscellaneous information for pca
# Adding reduction tsne
# Adding cell embeddings for tsne
# Adding miscellaneous information for tsne
# Adding reduction umap
# Adding cell embeddings for umap
# Adding miscellaneous information for umap
# Adding command information
# Adding cell-level metadata
# Adding miscellaneous information
# Adding tool-specific results
onecut3 &lt;- subset(onecut, subset = Onecut3 &gt; 0)</code></pre>
<p>Derive and filter matrix of Onecut3</p>
<pre class="r"><code>mtx_oc3 &lt;-
    onecut3 %&gt;%
    GetAssayData(&quot;data&quot;, &quot;RNA&quot;) %&gt;%
    as.data.frame() %&gt;%
    t()
rownames(mtx_oc3) &lt;- colnames(onecut3)

# Filter features
filt_low_genes &lt;-
    colSums(mtx_oc3) %&gt;%
    .[. &gt; quantile(., 0.4)] %&gt;%
    names()
mtx_oc3 %&lt;&gt;% .[, filt_low_genes]

min_filt_vector &lt;-
    mtx_oc3 %&gt;%
    as_tibble() %&gt;%
    select(all_of(filt_low_genes)) %&gt;%
    summarise(across(.fns = ~ quantile(.x, .1))) %&gt;%
    as.list %&gt;%
    map(as.double) %&gt;%
    simplify %&gt;%
    .[colnames(mtx_oc3)]

# Prepare table of intersection sets analysis
content_mtx_oc3 &lt;-
    (mtx_oc3 &gt; min_filt_vector) %&gt;%
    as_tibble() %&gt;%
    mutate_all(as.numeric)</code></pre>
<p>Correlation analysis visualisation between different genes</p>
<pre class="r"><code>p_corrs &lt;- list(
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        x = Onecut3,  y = Trh, xfill = &quot;#ffc400&quot;, yfill = &quot;#e22ee2&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        x = Slc32a1,  y = Onecut3,  xfill = &quot;#0000da&quot;,   yfill = &quot;#ffc400&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        x = Th,  y = Onecut3,  xfill = &quot;#006eff&quot;,   yfill = &quot;#ffc400&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        x = Slc32a1,  y = Trh, xfill = &quot;#0000da&quot;,   yfill = &quot;#e22ee2&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        x = Th,  y = Trh, xfill = &quot;#006eff&quot;,   yfill = &quot;#e22ee2&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        x = Slc32a1,  y = Th,  xfill = &quot;#0000da&quot;,   yfill = &quot;#006eff&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        y = Slc32a1,  x = Onecut3,  yfill = &quot;#0000da&quot;,   xfill = &quot;#ffc400&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        y = Th,  x = Onecut3,  yfill = &quot;#006eff&quot;,   xfill = &quot;#ffc400&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        y = Slc17a6,  x = Onecut3,  yfill = &quot;#ff0000&quot;,   xfill = &quot;#ffc400&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        y = Gad1,  x = Onecut3,  yfill = &quot;#a50202&quot;,    xfill = &quot;#ffc400&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        y = Gad2,  x = Onecut3,  yfill = &quot;#4002a5&quot;,    xfill = &quot;#ffc400&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        y = Onecut2,  x = Onecut3,  yfill = &quot;#6402a5&quot;,    xfill = &quot;#ffc400&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        y = Nav1,  x = Onecut3,  yfill = &quot;#2502a5&quot;,    xfill = &quot;#ffc400&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        y = Nav2,  x = Onecut3,  yfill = &quot;#4002a5&quot;,    xfill = &quot;#ffc400&quot;),
    ggstatsplot::ggscatterstats(as.data.frame(mtx_oc3),
        y = Trio,  x = Onecut3,  yfill = &quot;#2502a5&quot;,    xfill = &quot;#ffc400&quot;)
)
# Registered S3 method overwritten by &#39;ggside&#39;:
#   method from   
#   +.gg   ggplot2
n_corrs &lt;- list(
    &quot;oc3-rna-data-Onecut3-Trh&quot;,
    &quot;oc3-rna-data-Slc32a1-Onecut3&quot;,
    &quot;oc3-rna-data-Th-Onecut3&quot;,
    &quot;oc3-rna-data-Slc32a1-Trh&quot;,
    &quot;oc3-rna-data-Th-Trh&quot;,
    &quot;oc3-rna-data-Slc32a1-Th&quot;,
    &quot;oc3-rna-data-Onecut3-Slc32a1&quot;,
    &quot;oc3-rna-data-Onecut3-Th&quot;,
    &quot;oc3-rna-data-Onecut3-Slc17a6&quot;,
    &quot;oc3-rna-data-Onecut3-Gad1&quot;,
    &quot;oc3-rna-data-Onecut3-Gad2&quot;,
    &quot;oc3-rna-data-Onecut3-Onecut2&quot;,
    &quot;oc3-rna-data-Onecut3-Nav1&quot;,
    &quot;oc3-rna-data-Onecut3-Nav2&quot;,
    &quot;oc3-rna-data-Onecut3-Trio&quot;
)

walk2(n_corrs, p_corrs, save_my_plot, type = &quot;stat-corr-plt&quot;)
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p>Visualise intersections sets that we are going to use (highlighted)</p>
<pre class="r"><code>upset(as.data.frame(content_mtx_oc3),
  order.by = &quot;freq&quot;,
  sets.x.label = &quot;Number of cells&quot;,
  text.scale = c(2, 1.6, 2, 1.3, 2, 3),
  nsets = 15,
  sets = c(&quot;Gad1&quot;, &quot;Gad2&quot;, &quot;Slc32a1&quot;, &quot;Slc17a6&quot;),
  queries = list(
    list(
      query = intersects,
      params = list(&quot;Gad1&quot;, &quot;Gad2&quot;, &quot;Slc32a1&quot;),
      active = T
    ),
    list(
      query = intersects,
      params = list(&quot;Slc17a6&quot;),
      active = T
    )
  ),
  nintersects = 60,
  empty.intersections = &quot;on&quot;
)</code></pre>
<p><img src="figure/dynamic_oc3-pop.Rmd/plot-upset-oc3-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>Regroup factor by stages for more balanced groups</p>
<pre class="r"><code>onecut3$age %&gt;% forcats::fct_count()
# # A tibble: 6 × 2
#   f         n
#   &lt;fct&gt; &lt;int&gt;
# 1 E15     156
# 2 E17     104
# 3 P00      77
# 4 P02      74
# 5 P10     174
# 6 P23       7
onecut3$stage &lt;-
  onecut3$age %&gt;%
  forcats::fct_collapse(`Embrionic day 15` = &quot;E15&quot;,
                        `Embrionic day 17` = &quot;E17&quot;,
                        Neonatal = c(&quot;P00&quot;, &quot;P02&quot;),
                        Postnatal = c(&quot;P10&quot;, &quot;P23&quot;))
onecut3$stage %&gt;% forcats::fct_count()
# # A tibble: 4 × 2
#   f                    n
#   &lt;fct&gt;            &lt;int&gt;
# 1 Embrionic day 15   156
# 2 Embrionic day 17   104
# 3 Neonatal           151
# 4 Postnatal          181</code></pre>
<p>Make subset of stable neurons</p>
<pre class="r"><code>onecut3$gaba_status &lt;-
  content_mtx_oc3 %&gt;%
  select(Gad1, Gad2, Slc32a1) %&gt;%
  mutate(gaba = if_all(.fns = ~ .x &gt; 0)) %&gt;%
  .$gaba

onecut3$gaba_occurs &lt;-
  content_mtx_oc3 %&gt;%
  select(Gad1, Gad2, Slc32a1) %&gt;%
  mutate(gaba = if_any(.fns = ~ .x &gt; 0)) %&gt;%
  .$gaba

onecut3$glut_status &lt;-
  content_mtx_oc3 %&gt;%
  select(Slc17a6) %&gt;%
  mutate(glut = Slc17a6 &gt; 0) %&gt;%
  .$glut

oc3_fin &lt;-
  subset(onecut3,
    cells = union(
      WhichCells(onecut3,
        expression = gaba_status == TRUE &amp; glut_status == FALSE),
      WhichCells(onecut3,
        expression = glut_status == TRUE &amp; gaba_occurs == FALSE)))</code></pre>
<div id="check-contingency-tables-for-neurotransmitter-signature" class="section level2">
<h2>Check contingency tables for neurotransmitter signature</h2>
<pre class="r"><code>oc3_fin@meta.data %&gt;%
  janitor::tabyl(glut_status, gaba_status)
#  glut_status FALSE TRUE
#        FALSE     0  102
#         TRUE    92    0</code></pre>
<p>By age</p>
<pre class="r"><code>oc3_fin@meta.data %&gt;%
  janitor::tabyl(age, gaba_status)
#  age FALSE TRUE
#  E15    29    9
#  E17    12    4
#  P00    12    6
#  P02    16    2
#  P10    21   81
#  P23     2    0</code></pre>
<p>By stage</p>
<pre class="r"><code>oc3_fin@meta.data %&gt;%
  janitor::tabyl(stage, gaba_status)
#             stage FALSE TRUE
#  Embrionic day 15    29    9
#  Embrionic day 17    12    4
#          Neonatal    28    8
#         Postnatal    23   81</code></pre>
<p>Make splits of neurons by neurotransmitter signature</p>
<pre class="r"><code>oc3_fin$status &lt;- oc3_fin$gaba_status %&gt;%
  if_else(true = &quot;GABAergic&quot;,
          false = &quot;glutamatergic&quot;)
Idents(oc3_fin) &lt;- &quot;status&quot;

## Split on basis of neurotrans and test for difference
oc3_fin_neurotrans &lt;- SplitObject(oc3_fin, split.by = &quot;status&quot;)

## Split on basis of age and test for difference
oc3_fin_ages       &lt;- SplitObject(oc3_fin, split.by = &quot;age&quot;)</code></pre>
</div>
<div id="dotplots-grouped-by-age" class="section level2">
<h2>DotPlots grouped by age</h2>
<p>Expression of GABA receptors in GABAergic Onecut3 positive cells</p>
<pre class="r"><code>DotPlot(object = oc3_fin_neurotrans$GABAergic,
        features = gabar,
        group.by = &quot;age&quot;,
        cols = c(&quot;#adffff&quot;, &quot;#0084ff&quot;),
        col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/dynamic_oc3-pop.Rmd/plot-dotplot-age-gabar-gaba-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>Expression of GABA receptors in glutamatergic Onecut3 positive cells</p>
<pre class="r"><code>DotPlot(object = oc3_fin_neurotrans$glutamatergic,
        features = gabar,
        group.by = &quot;age&quot;,
        cols = c(&quot;#ffc2c2&quot;, &quot;#ff3c00&quot;),
        col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/dynamic_oc3-pop.Rmd/plot-dotplot-age-gabar-glu-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>Expression of glutamate receptors in GABAergic Onecut3 positive cells</p>
<pre class="r"><code>DotPlot(object = oc3_fin_neurotrans$GABAergic,
        features = glutr,
        group.by = &quot;age&quot;,
        cols = c(&quot;#adffff&quot;, &quot;#0084ff&quot;),
        col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/dynamic_oc3-pop.Rmd/plot-dotplot-age-glur-gaba-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>Expression of glutamate receptors in glutamatergic Onecut3 positive cells</p>
<pre class="r"><code>DotPlot(object = oc3_fin_neurotrans$glutamatergic,
        features = glutr,
        group.by = &quot;age&quot;,
        cols = c(&quot;#ffc2c2&quot;, &quot;#ff3c00&quot;),
        col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/dynamic_oc3-pop.Rmd/plot-dotplot-age-glur-glu-1.svg" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="dotplots-grouped-by-stage" class="section level2">
<h2>DotPlots grouped by stage</h2>
<p>Expression of GABA receptors in GABAergic Onecut3 positive cells</p>
<pre class="r"><code>DotPlot(object = oc3_fin_neurotrans$GABAergic,
        features = gabar,
        group.by = &quot;stage&quot;,
        cols = c(&quot;#adffff&quot;, &quot;#0084ff&quot;),
        col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/dynamic_oc3-pop.Rmd/plot-dotplot-stage-gabar-gaba-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>Expression of GABA receptors in glutamatergic Onecut3 positive cells</p>
<pre class="r"><code>DotPlot(object = oc3_fin_neurotrans$glutamatergic,
        features = gabar,
        group.by = &quot;stage&quot;,
        cols = c(&quot;#ffc2c2&quot;, &quot;#ff3c00&quot;),
        col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/dynamic_oc3-pop.Rmd/plot-dotplot-stage-gabar-glu-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>Expression of glutamate receptors in GABAergic Onecut3 positive cells</p>
<pre class="r"><code>DotPlot(object = oc3_fin_neurotrans$GABAergic,
        features = glutr,
        group.by = &quot;stage&quot;,
        cols = c(&quot;#adffff&quot;, &quot;#0084ff&quot;),
        col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/dynamic_oc3-pop.Rmd/plot-dotplot-stage-glur-gaba-1.svg" width="672" style="display: block; margin: auto;" /></p>
<p>Expression of glutamate receptors in glutamatergic Onecut3 positive cells</p>
<pre class="r"><code>DotPlot(object = oc3_fin_neurotrans$glutamatergic,
        features = glutr,
        group.by = &quot;stage&quot;,
        cols = c(&quot;#ffc2c2&quot;, &quot;#ff3c00&quot;),
        col.min = -1, col.max = 1
) + RotatedAxis()</code></pre>
<p><img src="figure/dynamic_oc3-pop.Rmd/plot-dotplot-stage-glur-glu-1.svg" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.1.2 (2021-11-01)
# Platform: x86_64-redhat-linux-gnu (64-bit)
# Running under: Rocky Linux 8.5 (Green Obsidian)
# 
# Matrix products: default
# BLAS/LAPACK: /usr/lib64/libopenblas-r0.3.12.so
# 
# locale:
#  [1] LC_CTYPE=C.UTF-8           LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
#  [1] Nebulosa_1.4.0        patchwork_1.1.1       UpSetR_1.4.0         
#  [4] SeuratDisk_0.0.0.9019 SeuratWrappers_0.3.0  SeuratObject_4.0.4   
#  [7] Seurat_4.0.6          sctransform_0.3.2     future_1.23.0        
# [10] zeallot_0.1.0         magrittr_2.0.1        forcats_0.5.1        
# [13] stringr_1.4.0         dplyr_1.0.7           purrr_0.3.4          
# [16] readr_2.1.1           tidyr_1.1.4           tibble_3.1.6         
# [19] ggplot2_3.3.5         tidyverse_1.3.1       here_1.0.1           
# [22] workflowr_1.7.0      
# 
# loaded via a namespace (and not attached):
#   [1] scattermore_0.7             R.methodsS3_1.8.1          
#   [3] coda_0.19-4                 ragg_1.2.1                 
#   [5] bit64_4.0.5                 knitr_1.37                 
#   [7] irlba_2.3.5                 DelayedArray_0.20.0        
#   [9] R.utils_2.11.0              data.table_1.14.2          
#  [11] rpart_4.1-15                RCurl_1.98-1.5             
#  [13] generics_0.1.1              BiocGenerics_0.40.0        
#  [15] callr_3.7.0                 cowplot_1.1.1              
#  [17] RANN_2.6.1                  correlation_0.7.1          
#  [19] bit_4.0.4                   tzdb_0.2.0                 
#  [21] spatstat.data_2.1-2         xml2_1.3.3                 
#  [23] lubridate_1.8.0             httpuv_1.6.5               
#  [25] SummarizedExperiment_1.24.0 assertthat_0.2.1           
#  [27] WRS2_1.1-3                  xfun_0.29                  
#  [29] hms_1.1.1                   jquerylib_0.1.4            
#  [31] evaluate_0.14               promises_1.2.0.1           
#  [33] fansi_1.0.0                 dbplyr_2.1.1               
#  [35] readxl_1.3.1                igraph_1.2.11              
#  [37] DBI_1.1.2                   htmlwidgets_1.5.4          
#  [39] reshape_0.8.8               spatstat.geom_2.3-1        
#  [41] kSamples_1.2-9              stats4_4.1.2               
#  [43] Rmpfr_0.8-7                 paletteer_1.4.0            
#  [45] ellipsis_0.3.2              ks_1.13.3                  
#  [47] backports_1.4.1             insight_0.15.0             
#  [49] deldir_1.0-6                MatrixGenerics_1.6.0       
#  [51] vctrs_0.3.8                 SingleCellExperiment_1.16.0
#  [53] Biobase_2.54.0              remotes_2.4.2              
#  [55] ROCR_1.0-11                 abind_1.4-5                
#  [57] cachem_1.0.6                withr_2.4.3                
#  [59] vroom_1.5.7                 mclust_5.4.9               
#  [61] goftest_1.2-3               cluster_2.1.2              
#  [63] lazyeval_0.2.2              crayon_1.4.2               
#  [65] hdf5r_1.3.5                 labeling_0.4.2             
#  [67] pkgconfig_2.0.3             SuppDists_1.1-9.7          
#  [69] GenomeInfoDb_1.30.0         nlme_3.1-153               
#  [71] statsExpressions_1.2.0      rlang_0.4.12               
#  [73] globals_0.14.0              lifecycle_1.0.1            
#  [75] miniUI_0.1.1.1              MatrixModels_0.5-0         
#  [77] modelr_0.1.8                rsvd_1.0.5                 
#  [79] cellranger_1.1.0            rprojroot_2.0.2            
#  [81] polyclip_1.10-0             matrixStats_0.61.0         
#  [83] lmtest_0.9-39               datawizard_0.2.2           
#  [85] Matrix_1.3-4                mc2d_0.1-21                
#  [87] zoo_1.8-9                   reprex_2.0.1               
#  [89] whisker_0.4                 ggridges_0.5.3             
#  [91] processx_3.5.2              png_0.1-7                  
#  [93] viridisLite_0.4.0           PMCMRplus_1.9.3            
#  [95] parameters_0.16.0           bitops_1.0-7               
#  [97] getPass_0.2-2               R.oo_1.24.0                
#  [99] KernSmooth_2.23-20          multcompView_0.1-8         
# [101] parallelly_1.30.0           S4Vectors_0.32.3           
# [103] scales_1.1.1                memoise_2.0.1              
# [105] plyr_1.8.6                  ica_1.0-2                  
# [107] zlibbioc_1.40.0             compiler_4.1.2             
# [109] RColorBrewer_1.1-2          fitdistrplus_1.1-6         
# [111] snakecase_0.11.0            cli_3.1.0                  
# [113] XVector_0.34.0              listenv_0.8.0              
# [115] pbapply_1.5-0               ps_1.6.0                   
# [117] ggside_0.2.0                MASS_7.3-54                
# [119] mgcv_1.8-38                 tidyselect_1.1.1           
# [121] stringi_1.7.6               textshaping_0.3.6          
# [123] highr_0.9                   yaml_2.2.1                 
# [125] ggrepel_0.9.1               grid_4.1.2                 
# [127] sass_0.4.0                  tools_4.1.2                
# [129] future.apply_1.8.1          parallel_4.1.2             
# [131] rstudioapi_0.13             git2r_0.29.0               
# [133] janitor_2.1.0               gridExtra_2.3              
# [135] farver_2.1.0                Rtsne_0.15                 
# [137] digest_0.6.29               BiocManager_1.30.16        
# [139] shiny_1.7.1                 pracma_2.3.6               
# [141] BWStest_0.2.2               Rcpp_1.0.8                 
# [143] GenomicRanges_1.46.1        broom_0.7.11               
# [145] BayesFactor_0.9.12-4.3      performance_0.8.0          
# [147] later_1.3.0                 RcppAnnoy_0.0.19           
# [149] httr_1.4.2                  effectsize_0.5             
# [151] colorspace_2.0-2            rvest_1.0.2                
# [153] fs_1.5.2                    tensor_1.5                 
# [155] reticulate_1.22             IRanges_2.28.0             
# [157] splines_4.1.2               uwot_0.1.11                
# [159] rematch2_2.1.2              spatstat.utils_2.3-0       
# [161] systemfonts_1.0.3           plotly_4.10.0              
# [163] xtable_1.8-4                gmp_0.6-2.1                
# [165] jsonlite_1.7.3              R6_2.5.1                   
# [167] pillar_1.6.4                htmltools_0.5.2            
# [169] mime_0.12                   glue_1.6.0                 
# [171] fastmap_1.1.0               codetools_0.2-18           
# [173] mvtnorm_1.1-3               utf8_1.2.2                 
# [175] lattice_0.20-45             bslib_0.3.1                
# [177] spatstat.sparse_2.1-0       leiden_0.3.9               
# [179] gtools_3.9.2                survival_3.2-13            
# [181] rmarkdown_2.11              munsell_0.5.0              
# [183] GenomeInfoDbData_1.2.7      ggstatsplot_0.9.0          
# [185] haven_2.4.3                 reshape2_1.4.4             
# [187] gtable_0.3.0                bayestestR_0.11.5          
# [189] spatstat.core_2.3-2</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
